pageserver/src/bin/pageserver.rs

```diff
@@ -223,13 +223,13 @@ fn initialize_config(
         }
     };
 
-    let config: toml_edit::Document = match std::fs::File::open(cfg_file_path) {
+    let config: pageserver_api::config::ConfigToml = match std::fs::File::open(cfg_file_path) {
         Ok(mut f) => {
             let md = f.metadata().context("stat config file")?;
             if md.is_file() {
                 let mut s = String::new();
                 f.read_to_string(&mut s).context("read config file")?;
-                s.parse().context("parse config file toml")?
+                toml_edit::de::from_str(&s).context("deserialize config file toml")?
```

Comment:
I think here we could use the the serde-whatever-track-depth (I'll look this up soon).

Reply:
At least the error now looks good:

```
Error: deserialize config toml

Caused by:
    tenant_config.gc_period: TOML parse error at line 11, column 13
       |
    11 | gc_period = "0"
       |             ^^^
    invalid value: string "0", expected a duration


Stack backtrace:
   0: pageserver::initialize_config
             at /home/joonas/src/neon/work/pageserver/src/bin/pageserver.rs:229:23
      pageserver::main
             at /home/joonas/src/neon/work/pageserver/src/bin/pageserver.rs:95:16
```

Without we get:

```
Error: deserialize config toml

Caused by:
    TOML parse error at line 11, column 13
       |
    11 | gc_period = "0"
       |             ^^^
    invalid value: string "0", expected a duration
```

which is strictly worse.

So I guess I need to check the other uses next...

Reply:
Oki, so the ` impl TryFrom<toml_edit::Item> for TenantConfOpt` in pageserver/src/tenant/config.rs:444 is probably/hopefully dead code now.

I think it's how the deserialization takes a different route, or maybe it's working around some annoyance which has gotten fixed in toml_edit. I remember adding that for a good reason...

Reply:
https://github.com/neondatabase/neon/issues/8918